spec/beaker/dsl/roles_spec.rb

```diff
@@ -163,6 +163,14 @@ class ClassMixedWithDSLRoles
       agent1[:type] = 'aio-foss'
       expect( subject.aio_version?(agent1) ).to be === true
     end
+    it 'can take an empty string for pe_ver' do
+      agent1[:pe_ver] = ''
+      expect{ subject.aio_version?(agent1) }.not_to raise_error
+    end
+    it 'can take an empty string for FOSS version' do
+      agent1[:version] = ''
+      expect{ subject.aio_version?(agent1) }.not_to raise_error
+    end
```

Comment:
@rick added truth table-style tests.  Let me know what you think when you get some time.